src/hera/workflows/script.py

```diff
@@ -760,6 +755,27 @@ def script_wrapper(func: Callable[FuncIns, FuncR]) -> Callable:
         def task_wrapper(*args, **kwargs) -> Union[FuncR, Step, Task, None]:
             """Invokes a `Script` object's `__call__` method using the given SubNode (Step or Task) args/kwargs."""
             if _context.active:
+                if len(args) == 1 and isinstance(args[0], (InputV1, InputV2)):
+                    arguments = args[0]._get_as_arguments()
+                    arguments_list = [
+                        *(arguments.artifacts or []),
+                        *(arguments.parameters or []),
+                    ]
+
+                    subnode = s.__call__(arguments=arguments_list, **kwargs)
+                    if not subnode:
+                        raise SyntaxError("Cannot use Pydantic I/O outside of a DAG, Steps or Parallel context")
+
+                    output_class = inspect.signature(func).return_annotation
+                    if not output_class or output_class is NoneType:
+                        return None
+
+                    if not issubclass(output_class, (OutputV1, OutputV2)):
```

Comment:
I'm not sure this should be a requirement, especially for gradual migrations and issues like https://github.com/argoproj-labs/hera/issues/1191#issue-2507988901

Reply:
I'm not sure what to return that would match the type signature though. Perhaps this can be addressed in a later PR?

Reply:
Setting the `subnode._build_obj` means users can do `my_task.parameter_name` when passes values between tasks - that behaviour should really only be available for the new decorator contexts as it's very experimental and unproven (see #1162). I think this PR is mixing up the two features which should be kept separate IMO, so I'm struggling to understand the motivation of the PR.

Reply:
Moved the new behaviour into the decorator_syntax experimental feature for now.